### ISSUE TITLE
Getting `ProviderNotFoundException` while comparing .aar files

### ISSUE THREAD

**Issue by AnwarShahriar** [JakeWharton/diffuse#68]
Was trying to find a diff of one of the privately published library in my team. I'm getting the following stacktrace.

```log
> ./diffuse diff --aar catlib-2.2.1.aar catlib-2.2.4.aar
Exception in thread "main" java.nio.file.ProviderNotFoundException: Provider not found
	at java.nio.file.FileSystems.newFileSystem(FileSystems.java:407)
	at com.jakewharton.diffuse.ExtensionsKt.asZipFileSystem(extensions.kt:28)
	at com.jakewharton.diffuse.ExtensionsKt.asZipFileSystem$default(extensions.kt:28)
	at com.jakewharton.diffuse.io.ZipKt.toZip(Zip.kt:93)
	at com.jakewharton.diffuse.io.PathInput.toZip(Input.kt:43)
	at com.jakewharton.diffuse.Jar$Companion.parse(Jar.kt:21)
	at com.jakewharton.diffuse.Aar$Companion.parse(Aar.kt:29)
	at com.jakewharton.diffuse.DiffCommand$inputOptions$2.parse(diffuse.kt:77)
	at com.jakewharton.diffuse.DiffCommand.run(diffuse.kt:133)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:139)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:14)
	at com.github.ajalt.clikt.core.CliktCommand.parse(CliktCommand.kt:215)
	at com.github.ajalt.clikt.core.CliktCommand.parse$default(CliktCommand.kt:212)
	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:230)
	at com.jakewharton.diffuse.Diffuse.main(diffuse.kt:40)
```

**Comment by JakeWharton** [comment]
Unfortunately, due to a bug/missing feature in Java's implementation of zip filesystems, you have to use Java 11 or newer for aars or aabs.

**Comment by felipeska** [comment]
I'm comparing two aar and this error happens:

<img width="974" alt="Screen Shot 2020-05-20 at 1 06 06 PM" src="https://user-images.githubusercontent.com/613443/82481394-db1d6800-9a9a-11ea-9b9b-ae34a851bab2.png">

**Comment by JakeWharton** [comment]
Are these aars public? Can you reproduce it with public aars? You can grab two ThreeTenABP aars or something and try: https://search.maven.org/artifact/com.jakewharton.threetenabp/threetenabp

**Comment by felipeska** [comment]
Trying with public aars:

<img width="1087" alt="Screen Shot 2020-05-20 at 1 17 53 PM" src="https://user-images.githubusercontent.com/613443/82482512-98f52600-9a9c-11ea-9dd2-0b71ef0cbe2b.png">

**Comment by JakeWharton** [comment]
Fun. I will try to reproduce!

**Comment by cemore2048** [comment]
Is it still the case that we need to be using Java 11? FWIW, I'm getting the same error trying to diff 2 aars

**Comment by JakeWharton** [comment]
11 is the minimum supported version, yes. If you are using 11 and seeing the error then it is likely this bug that I haven't had time to look into.

**Comment by JakeWharton** [comment]
I can reproduce this with 11.0.1+13 but not with 11.0.8+10. Going to close since it's a JDK problem that seems to have been fixed.

**Comment by JakeWharton** [comment]
For posterity:
```
$ java -version
openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.8+10, mixed mode)

$ java -jar diffuse/build/libs/diffuse-0.2.0-SNAPSHOT-binary.jar diff --aar ~/Downloads/threetenabp-1.2.3.aar ~/Downloads/threetenabp-1.2.4.aar
OLD: threetenabp-1.2.3.aar
NEW: threetenabp-1.2.4.aar

 AAR      │ old       │ new     │ diff
──────────┼───────────┼─────────┼────────
      jar │   1.9 KiB │ 1.9 KiB │    0 B
 manifest │     230 B │   230 B │    0 B
    other │     457 B │   457 B │    0 B
──────────┼───────────┼─────────┼────────
    total │ 102.8 KiB │ 103 KiB │ +202 B


 JAR     │ old │ new │ diff
─────────┼─────┼─────┼───────────
 classes │   2 │   2 │ 0 (+0 -0)
 methods │  22 │  22 │ 0 (+0 -0)
  fields │   3 │   3 │ 0 (+0 -0)


=================
====   AAR   ====
=================

 size      │ diff   │ path
───────────┼────────┼───────────────────────────────────
 100.4 KiB │ +202 B │ ∆ assets/org/threeten/bp/TZDB.dat
───────────┼────────┼───────────────────────────────────
 100.4 KiB │ +202 B │ (total)

$ ~/Downloads/zulu11.2.3-jdk11.0.1-macosx_x64/bin/java -version
openjdk version "11.0.1" 2018-10-16 LTS
OpenJDK Runtime Environment Zulu11.2+3 (build 11.0.1+13-LTS)
OpenJDK 64-Bit Server VM Zulu11.2+3 (build 11.0.1+13-LTS, mixed mode)

$ ~/Downloads/zulu11.2.3-jdk11.0.1-macosx_x64/bin/java -jar diffuse/build/libs/diffuse-0.2.0-SNAPSHOT-binary.jar diff --aar ~/Downloads/threetenabp-1.2.3.aar ~/Downloads/threetenabp-1.2.4.aar
Exception in thread "main" java.nio.file.ProviderNotFoundException: Provider not found
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:430)
	at com.jakewharton.diffuse.io.PathInput.toZip(Input.kt:24)
	at com.jakewharton.diffuse.format.Jar$Companion.parse(Jar.kt:1)
	at com.jakewharton.diffuse.format.Aar$Companion.parse(Aar.kt:4)
	at com.jakewharton.diffuse.DiffCommand.run(diffuse.kt:83)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:2095)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:2201)
	at com.github.ajalt.clikt.core.CliktCommand.parse$default(CliktCommand.kt:408)
	at com.jakewharton.diffuse.Diffuse.main(diffuse.kt:23453)
```

**Comment by d-txy** [comment]
This is not supported before JDK 12 via that specific constructor (Path, ClassLoader)

This was fixed in JDK12, with commit [196c20c0d14d99cc08fae64a74c802b061231a41](https://github.com/AdoptOpenJDK/openjdk-jdk/commit/196c20c0d14d99cc08fae64a74c802b061231a41)

The offending code was in ZipFileSystemProvider in JDK 11 and earlier:

https://stackoverflow.com/questions/44459152/provider-not-found-exception-when-creating-a-filesystem-for-my-zip
